### PR TITLE
use dex in HA mode

### PR DESCRIPTION
**Pull request by anvddriesch**
Towards https://github.com/giantswarm/roadmap/issues/797
## Checklist

- [x] Update CHANGELOG.md

### CODE REVIEW

**Comment by marians** [comment]
So was the decision to not make it configurable?

**Comment by PrashantR31** [comment]
> So was the decision to not make it configurable?

My bad, I skipped this in the list of actions: It is to make it configurable with a minimum of 2 replicas.

**Comment by anvddriesch** [comment]
I personally think this should be fine. Since we are using a 'preferred' affinity, both pods would be scheduled on the same node if there is only one.
I don't see why we would need more than two to be honest.

File: helm/dex-app/templates/dex/deployment.yaml [hunk 1]
```diff
@@ -20,6 +20,15 @@ spec:
         {{- include "dex.labels.common" . | nindent 8 }}
       annotations:
     spec:
+      affinity:
```
Comment (anvddriesch) [inline helm/dex-app/templates/dex/deployment.yaml:23]: This affinity term means that we try to schedule each replica on a separate node if possible.

Comment (PrashantR31) [reply helm/dex-app/templates/dex/deployment.yaml:23]: Okie, I understand now and makes sense. Thanks @anvddriesch

**Review by PrashantR31 — APPROVED** [review]

**Review by marians — APPROVED** [review]

**Comment by marians** [comment]
All right, fine with me!

**Pull request merged**